File: Solutions/1855. Maximum Distance Between a Pair of Values/1855. Maximum Distance Between a Pair of Values.java
class Solution {
    public int maxDistance(int[] nums1, int[] nums2) {
        int res=0;
        for(int i=0;i<nums1.length;++i){
            int tmp=bs(nums1[i],nums2);
            if(tmp<i)
                continue;
            if(res<tmp-i)
                res=tmp-i;
        }
        return res;
    }
    private int bs(int x,int[] nums){
        int l=0,r=nums.length-1;
        while(l<=r){
            int mid=(l+r)/2;
            if(x>nums[mid])
                r=mid-1;
            else //x>=nums[mid]
                l=mid+1;
        }
        return r;
    }
}

class Solution {
    public int maxDistance(int[] nums1, int[] nums2) {
        int i=0,j=0,n1=nums1.length,n2=nums2.length,res=0;
        while(i<n1 && j<n2){
            if(nums1[i]>nums2[j])
                ++i;
            else
                res=Math.max(res,j++-i);
        }
        return res;
    }
}

File: Solutions/1855. Maximum Distance Between a Pair of Values/1855. Maximum Distance Between a Pair of Values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i,j,n1,n2,res=0,0,len(nums1),len(nums2),0
        while i<n1 and j<n2:
            if nums1[i]>nums2[j]:
                i+=1
            else:
                res=max(res,j-i)
                j+=1
        return res